datawolf-executor-commandline/src/main/java/edu/illinois/ncsa/datawolf/executor/commandline/CommandLineExecutor.java

```diff
@@ -369,11 +376,11 @@ public void execute(File cwd) throws AbortException, FailedException {
                     ds.setTitle(step.getTool().getOutput(impl.getCaptureStdOut()).getTitle());
                     ds.setCreator(execution.getCreator());
 
+                    ds = datasetDao.save(ds);
```

Comment:
For external file storage like IN-CORE where the dataset is created and then files are attached in a separate operation, we need to save the dataset first. Otherwise, the dataset won't exist yet and file storage will fail.